tests/audit-scanner.bats

```diff
@@ -51,21 +53,88 @@ setup() {
+    testing_namespace_uid=$(kubectl get ns testing-audit-scanner -o=json | jq -r ".metadata.uid")
+    local report=$(kubectl get clusterpolicyreports "$testing_namespace_uid" -o json | jq -ec)
+    echo "$report" | jq -e '.summary.pass == 0'
+    echo "$report" | jq -e '.summary.fail == 1'
+    echo "$report" | jq -e '[.results[] | select(.policy == "clusterwide-safe-labels")] | all(.result == "fail")'
+    echo "$report" | jq '[.results[] | select(.policy == "clusterwide-psa-label-enforcer-policy")] | empty'
+
+    default_namespace_uid=$(kubectl get ns default -o=json | jq -r ".metadata.uid")
+    local report=$(kubectl get clusterpolicyreports "$default_namespace_uid" -o json | jq -ec)
+    echo "$report" | jq -e '.summary.pass == 1'
+    echo "$report" | jq -e '.summary.fail == 0'
+    echo "$report" | jq -e '[.results[] | select(.result=="pass")] | all'
+}
+
+@test "[Audit Scanner] Update namespaced policy reports after remove one policy" {
 
     unprivileged_pod_uid=$(kubectl get pods nginx-unprivileged  -o=json | jq -r ".metadata.uid")
     local report=$(kubectl get policyreports "$unprivileged_pod_uid" -o json | jq -ec)
     echo "$report" | jq -e '.summary.fail == 0'
     echo "$report" | jq -e '.summary.pass == 1'
     echo "$report" | jq -e '[.results[] | select(.policy == "clusterwide-privileged-pods") | .result=="pass"] | all'
+    echo "$report" | jq '[.results[] | select(.policy == "clusterwide-safe-labels-for-pods")] | empty'
+
+    privileged_pod_uid=$(kubectl get pods nginx-privileged  -o=json | jq -r ".metadata.uid")
+    local report=$(kubectl get policyreports "$privileged_pod_uid" -o json | jq -ec)
+    echo "$report" | jq -e '.summary.fail == 1'
+    echo "$report" | jq -e '.summary.pass == 0'
+    echo "$report" | jq -e '[.results[] | select(.policy == "clusterwide-privileged-pods") | .result=="fail"] | all'
+    echo "$report" | jq '[.results[] | select(.policy == "clusterwide-safe-labels-for-pods")] | empty'
 }
 
-teardown_file() {
+@test "[Audit Scanner] Delete all policies and retrigger audit scanner job" {
     kubectl delete -f $RESOURCES_DIR/privileged-pod-policy.yaml
-    kubectl delete -f $RESOURCES_DIR/namespace-label-propagator-policy.yaml
     kubectl delete -f $RESOURCES_DIR/safe-labels-namespace.yaml
-    kubectl delete ns testing-audit-scanner
-    kubectl delete pod nginx-privileged nginx-unprivileged
-    kubectl delete jobs -n kubewarden testing
+    wait_for_default_policy_server_rollout
+
+    kubectl delete --ignore-not-found job testing --namespace $NAMESPACE
+    run kubectl create job  --from=cronjob/audit-scanner testing  --namespace $NAMESPACE
+    assert_output -p "testing created"
+    kubectl wait --for=condition="Complete" job testing --namespace $NAMESPACE
+}
+
+@test "[Audit Scanner] Delete namespaced policy reports after all relevant policies" {
+    retry "kubectl get --ignore-not-found policyreport -A -o json | jq -e '.items[] | empty'"
```

Comment:
Do you mind moving this check to `Delete all policies and retrigger audit scanner job` above?

If the change is caused by previous test I think it would be more obvious if check is part of that instead of separate test one-liner.